### PR TITLE
Option to fully close the inventory

### PR DESCRIPTION
**Pull request by HSGamer**

### CODE REVIEW

File: api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java [hunk 1]
```diff
@@ -189,6 +189,17 @@ public void closeInventory() {
...
 
+	/**
+	 * Closes the inventory if it's open.
+	 * @param force whether it'll force the inventory to be fully closed
+	 */
+	public void closeInventory(boolean force) {
+		if (force) {
+			preventClose = false;
+		}
+		closeInventory();
```
Comment (portlek) [inline api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java:200]: I think this `preventClose` should be return the old value if force is true like that:
```java
final boolean oldvalue = preventClose;
if (force) {
    preventClose = false;
}
closeInventory();
if (force) {
    preventClose = oldvalue;
}
```

Comment (HSGamer) [reply api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java:201]: ok

**Comment by WesJD** [comment]
Whats event the point of this?

**Comment by portlek** [comment]
> Whats event the point of this?

You able to close a GUI that prevents close.(if the force is true) It can be useful I think.

**Comment by WesJD** [comment]
[The inventory close listener](https://github.com/WesJD/AnvilGUI/blob/master/api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java#L245) already first checks if the inventory is open, and open is set to false before the client is ever sent anything to close the gui. Am really confused as to why this PR exists at all.

**Comment by HSGamer** [comment]
> [The inventory close listener](https://github.com/WesJD/AnvilGUI/blob/master/api/src/main/java/net/wesjd/anvilgui/AnvilGUI.java#L245) already first checks if `open` is true, and `open` is set to false before the client is ever sent anything to close the gui from `closeInventory`. Am really confused as to why this PR exists at all.

`open` will be set to true as you will open the inventory again if `preventClose` is true

**Comment by HSGamer** [comment]
I've tested again and it works without this pull request. Safe to close

**Pull request closed without merging**